### ISSUE TITLE
View events not logged correctly in Xamarin.Forms

### ISSUE THREAD

**Issue by velocitysystems** [MindscapeHQ/raygun4net#469]
**Expected**
When real user monitoring is in enabled in a Xamarin.Forms application, views should be tracked according to their page name i.e. "SamplePage" or [shell navigation route](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/shell/navigation) i.e. "animals/domestic/cats".

**Actual**
Instead, Raygun records the underlying view controller or activity responsible for rendering the page i.e. `ContentPageRenderer` (iOS), `MainActivity`(Android) or similar.

**Steps to Reproduce**
- Add Raygun to a blank Xamarin.Forms application
- Enable Real User Monitoring
- Navigate to one or more pages

**Recommendations**
Either: (a) Latch to Xamarin.Form navigation events to accurately track page views, or (b) provide an API surface to override the default view event observers for page load & navigation events on each platform.

An example would be something like:
```
RaygunClient.Current.OverrideViewEvents = true;
Xamarin.Forms.Application.Current.PageAppearing += (sender, e) => {
    RaygunClient.Current.RecordViewEvent(e.GetType().Name);
}
```

**Comment by QuantumNightmare** [comment]
Hey, thanks for posting this. Regardless of how we'd go about implementing this, we're currently thinking to stick with the current behaviour as the default with some kind of option to switch to this different naming (at least for an initial roll out of this change). We aren't picking this up right now, but this has been logged with the product team for further discussion and prioritization.

**Comment by velocitysystems** [comment]
@QuantumNightmare I'm currently working on this for an XF application and will post back in this thread with my findings.
I'm also a `Xamarin.Forms` contributor, so I'm very familiar with the internal workings of XF and MAUI.

**Comment by DM1145** [comment]
FWIW, I'm seeing much the same thing, making this "feature" of very limited utility.  Is there a way to set those values in C# so they are properly picked up in Error Reporting and RUM?

**Comment by DM1145** [comment]
Noticed that it is of limited use - unfortunately need a solution soon, so....

**Comment by QuantumNightmare** [comment]
Thanks for these few extra details. I've passed this on to the teams internally here. We currently don't have capacity to implement a solution here, but aiming to at least discuss it this week and see where we can fit it in.

**Comment by velocitysystems** [comment]
Thanks @QuantumNightmare - Look forward to hearing the result of your discussion.

**Comment by DM1145** [comment]
Same.  Thank you-

**Comment by QuantumNightmare** [comment]
In our discussion, it was pointed out that it's recommended to use the [raygun4xamarin](https://github.com/MindscapeHQ/raygun4xamarin) provider (if you aren't already) instead of this raygun4net provider. To be clear, using raygun4xamarin does not have any difference to the view tracking, but it's a more well written provider and where we'd prefer adding improvements to our Xamarin.Forms support. Apologies for not picking up on this earlier.

This task has been moved to our backlog, but we don't have a time frame of when we'd pick this up. In the meantime, feel free to open a PR if you have time to provide an implementation - preferably on the raygun4xamarin provider, but we could accept a PR here too. Alternatively, feel free to contune providing any details/ideas that come to mind that would help us with the implementation later on.

Here are our thoughts so far:
* Provide an option to disable just the existing view tracking hooks. This would be for cases where developers want their own custom view tracking logic, such as your suggestion of overriding PageRenderer. (Page load events can already manually be sent to Raygun via the `RaygunClient.Current.SendPulseTimingEvent` method in raygun4net or `RaygunClient.Current.SendTimingEvent` method in raygun4xamarin).
* Provide an option to automatically hook into and track the Xamarin.Forms.Application.Current.PageAppearing method - though that has limited usefulness due to not being able to get page load time from this. Unless there's a way to somehow combine the page names from this with the load times from the Activity lifecycle events for example.
* Provide an option to automatically hook into and track Xamarin.Forms.Shell.CurrentShell navigation events. To my understanding the application need to be implemented to use the shell navigation in the first place for this to be of use.
* If it's useful for multiple of these view tracking approaches to be in use simultaneously, then perhaps these options could be implemented as a flags enum.
* As mentioned earlier, we'd keep the existing behaviour as the default for the time being.

Let us know if this aligns with what would work for you or if I've got any details wrong.
